feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/6a870cc60c23f0332a2fe788.md

---
id: 6a870cc60c23f0332a2fe788
title: Step 2
challengeType: 20
dashedName: step-2
---

# --description--

Remember from a previous workshop that you can join strings together using the `+` operator. This is called concatenation.

Here's a reminder of how to do it:

```py
name = 'Alice'
print('Hello, ' + name)  # Hello, Alice
```

> Now, print `email` and concatenate the prefix `'Email: '` with it.

Reviewer: Reads in the order the camper actually writes it, and quotes the literal the way steps 5 onward do.

```suggestion
Now, print the result of concatenating the string `'Email: '` with the `email` variable.
```

> Now, print the result of concatenating the string `'Email: '` with the `email` variable.

Reviewer 2: ```suggestion
Now, print the result of concatenating the string `Email: ` with the `email` variable.
```

# --hints--

> You should print the result of concatenating the string `Email: ` with the `email` variable.

Reviewer: ```suggestion
You should print the result of concatenating the string `'Email: '` with the `email` variable.
```

> You should print the result of concatenating the string `'Email: '` with the `email` variable.

Reviewer 2: ```suggestion
You should print the result of concatenating the string `Email: ` with the `email` variable.
```

---
id: 6a870cc60c23f0332a2fe788
title: Step 2
challengeType: 20
dashedName: step-2
---

# --description--

Remember from a previous workshop that you can use an f-string to include a variable inside a string, by writing an `f` before the opening quote and wrapping the variable in `{}`.

Here's a reminder of how it works:

> ```python

Reviewer: Fence language is split across the workshop: description examples are fenced ` ```py ` in steps 3, 5, 8, 11, 13, 14, 15, 16 and 20, and ` ```python ` in steps 2, 4, 9 (both blocks), 10 (both blocks), 17 and 18. `py` is also what the rest of the Python curriculum uses, so it is the one to keep.

Locations to change: step 2 L14, step 4 L14, step 9 L12 and L20, step 10 L12 and L20, step 17 L14, step 18 L14. No suggestion block here because the replacement line is itself a fence.

name = 'Alice'

> print(f'Hello, {name}')  # Hello, Alice

Reviewer: Example output comments use three styles. Nine steps annotate with `  # Output: X` (8, 9, 10, 11, 13, 14, 15, 16, 20), three use a bare `  # X` (2 L16, 3 L16, 4 L16), and three use a single-space ` # X` (5 L14 and L15, 17 L16 and L17, 18 L16 and L17).

Standardizing on `  # Output: X` covers the majority and also matches PEP 8's two spaces before an inline comment. Left as a plain note rather than eight near-identical suggestions.

```

> Now, print `email` so that the output reads: `Email: alice.johnson@company.com`

Reviewer: Steps 19 and 21 open with "Now print"; this is the only "Now, print".

The trailing punctuation is the cross-step item: "so that the output reads" is followed by a colon and no period in steps 2, 3 and 5, a colon and a period in steps 6, 11, 13 and 16, and no colon with a period in steps 4, 9 and 15. Every hint line in the workshop uses the last of those, so aligning the descriptions to it is the smallest change.

Optional and workshop-wide: the print hints accept an f-string or `'label: ' + var`, but not `print('Email:', email)`, which produces byte-identical output and is a natural thing for a beginner to write. Verified rejected. The two-form pattern matches the rest of the Python curriculum, so this is a judgment call rather than a defect.

```suggestion
Now print `email` so that the output reads `Email: alice.johnson@company.com`.
```
